src/components/Tooltip.js

```diff
@@ -1,97 +1,116 @@
+// rotated 45 degrees, calculate the length of 1 side of the box.
+// a^2 + b^2 = c^2
+const hypotenuse = (0.25 ** 2 + 0.25 ** 2) ** 0.5;
+
+const El = styled.div``;
 const TooltipEl = styled.div`
   position: absolute;
   display: block;
   border-radius: 4px;
   overflow: visible;
   font-size: 0.75rem;
-  width: 6.25rem;
-  height: auto;
   padding: 0.625rem;
   z-index: ${Z_INDEXES.tooltip};
   color: ${PALETTE.white};
   background-color: rgba(41, 45, 62, 1);
+  pointer-events: none;
+  opacity: 0;
+  transition: opacity ${(props) => props.duration}ms ease-in-out;
+  ${({ isOpen }) =>
+    isOpen
+      ? `
+          opacity: 1;
+          pointer-events: all;
+        `
+      : ""}
 
   &::after {
     content: "";
     position: absolute;
-    width: 0.625rem;
-    height: 0.625rem;
+    width: ${hypotenuse}rem;
+    height: ${hypotenuse}rem;
     right: 1rem;
-    bottom: -4px;
+    bottom: -0.25rem;
     background-color: rgba(41, 45, 62, 1);
     transform: rotate(45deg);
+    // Not sure why but this makes it look nicer. Maybe fine units getting
+    // trimmed weirdly?? One of the square's corners appears to be cut off.
+    transform-origin: top;
   }
 `;
 
-export const Tooltip = ({
-  in: inProp,
-  duration = 200,
-  parentPosition,
-  children,
-}) => {
-  const [tooltipDimension, setTooltipDimension] = React.useState({
+export const Tooltip = ({ duration = 150, message, children }) => {
+  const wrapperRef = React.useRef();
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [size, setSize] = React.useState({
     width: 0,
     height: 0,
   });
-  const [tooltipPosition, setTooltipPosition] = React.useState({
-    left: 0,
-    top: 0,
-  });
 
-  const tooltipRef = React.useCallback((node) => {
-    if (node !== null) {
-      setTooltipDimension({
-        width: node.getBoundingClientRect().width,
-        height: node.getBoundingClientRect().height,
+  const tooltipRef = React.useRef();
+  const updateDimensions = React.useCallback(() => {
+    if (tooltipRef.current) {
+      const { width, height } = tooltipRef.current.getBoundingClientRect();
+      setSize({
+        width,
+        height,
       });
     }
   }, []);
 
   React.useLayoutEffect(() => {
-    setTooltipPosition({
-      left: parentPosition.right - tooltipDimension.width + 16,
-      top:
-        window.pageYOffset + parentPosition.top + tooltipDimension.height - 90,
-    });
-  }, [tooltipDimension, parentPosition]);
+    if (!isOpen || !tooltipRef.current) {
+      return;
+    }
+    const { right, top } = wrapperRef.current.getBoundingClientRect();
+    const newRight = window.innerWidth - right;
+    // We need height + some because of the extra space for the little pointer
+    // bit. That's .25rem away
+    const newTop = window.pageYOffset + top - (size.height + 4);
+    tooltipRef.current.style.right = `${newRight}px`;
+    tooltipRef.current.style.top = `${newTop}px`;
```

Comment:
I switched this to imperatively set the size instead of setting state to avoid another render pass.

The major change here is that I switched it from `left:` to `right:`, which lets us more easily have dynamically sized content without the position changing